apps/frontend/src/libs/components/header/libs/components/notifications-popover/notifications-popover.tsx

```diff
@@ -18,26 +21,46 @@ type Properties = {
 	onClose: () => void;
 };
 
+const PAGE_SIZE = 10;
+
 const NotificationsPopover = ({
 	children,
 	isOpened,
 	onClose,
 }: Properties): JSX.Element => {
 	const dispatch = useAppDispatch();
-
-	const { notifications } = useAppSelector(
+	const { dataStatus, notifications, notificationsTotalCount } = useAppSelector(
 		({ notifications }) => notifications,
 	);
 
-	const handleLoadNotifications = useCallback(() => {
-		void dispatch(notificationActions.loadAll());
-	}, [dispatch]);
+	const handleLoadNotifications = useCallback(
+		(page: number, pageSize: number) => {
+			void dispatch(notificationActions.loadAll({ page, pageSize }));
+		},
+		[dispatch],
+	);
+
+	const { hasNextPage, onNextPage, onPageReset } = useInfiniteScroll({
+		currentItemsCount: notifications.length,
+		onLoading: handleLoadNotifications,
+		pageSize: PAGE_SIZE,
+		totalItemsCount: notificationsTotalCount,
+	});
+
+	const { reference: sentinelReference } =
+		useIntersectionObserver<HTMLDivElement>({
+			isDisabled: !hasNextPage || dataStatus === "pending",
```

Comment:
use an enum here

Reply:
Now is using enum like the other places in code